tests/bmc_api_test.go

```diff
@@ -0,0 +1,747 @@
+
+	// Asserts
+	suite.Equal(string(jsonResult), string(jsonResponseBody))
+
+	// Verify
+	suite.verifyCalledOnce(expectationId)
+}
+
+func (suite *BmcApiTestSuite) TestGetSshKeys() {
+	// Generate payload
+	request, response := TestUtilsImpl{}.generatePayloadsFrom("bmcapi/sshkeys/sshkeys_get", "./payloads")
+
+	// Extract the response expectation id
+	expectationId := TestUtilsImpl{}.setupExpectation(request, response, 1)
+
+	// Operation Execution
+	result, _, _ := suite.Client.SSHKeysApi.SshKeysGet(suite.Ctx).Execute()
+
+	// Convert the result and response body to json strings
+	jsonResult, _ := json.Marshal(result)
+	jsonResponseBody, _ := json.Marshal(response.Body)
+
+	// Asserts
+	suite.Equal(string(jsonResult), string(jsonResponseBody))
+
+	// Verify
+	suite.verifyCalledOnce(expectationId)
+}
+
+func (suite *BmcApiTestSuite) TestCreateSshKey() {
+	// Generate payload
+	request, response := TestUtilsImpl{}.generatePayloadsFrom("bmcapi/sshkeys/sshkeys_post", "./payloads")
+
+	// Extract the response expectation id
+	expectationId := TestUtilsImpl{}.setupExpectation(request, response, 1)
+
+	// Prepare request body
+	body, _ := json.Marshal(request.Body.Json)
+	var sshKeyCreate bmcapi.SshKeyCreate
+	json.Unmarshal(body, &sshKeyCreate)
+
+	// Operation Execution
+	result, _, _ := suite.Client.SSHKeysApi.SshKeysPost(context.Background()).SshKeyCreate(sshKeyCreate).Execute()
+
+	// Convert the result and response body to json strings
+	jsonResult, _ := json.Marshal(result)
+	jsonResponseBody, _ := json.Marshal(response.Body)
+
+	// Asserts
+	suite.Equal(string(jsonResult), string(jsonResponseBody))
+
+	// Verify
+	suite.verifyCalledOnce(expectationId)
+}
+
+func (suite *BmcApiTestSuite) TestGetSshKeyById() {
+	// Generate payload
+	request, response := TestUtilsImpl{}.generatePayloadsFrom("bmcapi/sshkeys/sshkeys_get_by_id", "./payloads")
+
+	// Extract the response expectation id
+	expectationId := TestUtilsImpl{}.setupExpectation(request, response, 1)
+
+	// Extract the quotaId
+	sshKeyId := request.PathParameters["id"][0]
+
+	// Operation Execution
+	result, _, _ := suite.Client.SSHKeysApi.SshKeysSshKeyIdGet(context.Background(), sshKeyId).Execute()
+
+	// Convert the result and response body to json strings
+	jsonResult, _ := json.Marshal(result)
+	jsonResponseBody, _ := json.Marshal(response.Body)
+
+	// Asserts
+	suite.Equal(string(jsonResult), string(jsonResponseBody))
+
+	// Verify
+	suite.verifyCalledOnce(expectationId)
+}
+
+func (suite *BmcApiTestSuite) TestDeleteSshKeyById() {
+	// Generate payload
+	request, response := TestUtilsImpl{}.generatePayloadsFrom("bmcapi/sshkeys/sshkeys_delete_by_id", "./payloads")
+
+	// Extract the response expectation id
+	expectationId := TestUtilsImpl{}.setupExpectation(request, response, 1)
+
+	// Extract the quotaId
+	sshKeyId := request.PathParameters["id"][0]
+
+	// Operation Execution
+	result, _, _ := suite.Client.SSHKeysApi.SshKeysSshKeyIdDelete(context.Background(), sshKeyId).Execute()
```

Comment:
Was suite.ctx giving you issues when used here?

Reply:
Ah, forgot to replace it. Changed.